### ISSUE TITLE
Create an interface to scipy's minimize

### ISSUE THREAD

**Issue by matteobachetti** [sherpa/sherpa#170]
Dear developers,

It would be great if the user was able to compare the fitting results of `sherpa` to those of any of `scipy`'s minimization routines. There is an easy way, and it is making an interface to a single function, [`scipy.optimize.minimize`](http://docs.scipy.org/doc/scipy-0.16.1/reference/generated/scipy.optimize.minimize.html). This function already contains the possibility to call any of the minimization methods in `scipy`.

Thanks for this great software,
Matteo

**Comment by dtnguyen2** [comment]
If you look into the file sherpa/optmethods/__init__.py, you will find the
following lines:

###############################################################################

## from sherpa.optmethods.fminpowell import *
## from sherpa.optmethods.nmpfit import *

## from sherpa.optmethods.odrpack import odrpack
## from sherpa.optmethods.stogo import stogo
## from sherpa.optmethods.chokkan import chokkanlbfgs
## from sherpa.optmethods.odr import odrf77

## def myall( targ, arg ):
##     fubar = list( targ )
##     fubar.append( arg )
##     return tuple( fubar )

## __all__ = myall( __all__, 'Bobyqa' )
## __all__ = myall( __all__, 'Chokkan' )
## __all__ = myall( __all__, 'cppLevMar' )
## __all__ = myall( __all__, 'Dif_Evo' )
## __all__ = myall( __all__, 'MarLev' )
## __all__ = myall( __all__, 'MyMinim' )
## __all__ = myall( __all__, 'Nelder_Mead' )
## __all__ = myall( __all__, 'NMPFIT' )
## __all__ = myall( __all__, 'Newuoa' )
## __all__ = myall( __all__, 'Odr' )
## __all__ = myall( __all__, 'OdrPack' )
## __all__ = myall( __all__, 'PortChi' )
## __all__ = myall( __all__, 'PortFct' )
## __all__ = myall( __all__, 'ScipyPowell' )
## __all__ = myall( __all__, 'StoGo' )

## class Chokkan(OptMethod):
##     def __init__(self, name='chokkan'):
##         OptMethod.__init__(self, name, chokkanlbfgs)

## class cppLevMar(OptMethod):

##    def __init__(self, name='clevmar'):
##     OptMethod.__init__(self, name, optfcts.lmdif_cpp)

## class MyMinim(OptMethod):

##     def __init__(self, name='simplex'):
##     OptMethod.__init__(self, name, minim)

## class NMPFIT(OptMethod):
##     def __init__(self, name='pytools_nmpfit'):
##         OptMethod.__init__(self, name, nmpfit.pytools_nmpfit)

## class OdrPack(OptMethod):
##     def __init__(self, name='odrpack'):
##         OptMethod.__init__(self, name, odrpack)

## class Odr(OptMethod):
##     def __init__(self, name='odr'):
##         OptMethod.__init__(self, name, odrf77)

## class ScipyPowell(OptMethod):
##     def __init__(self, name='scipypowell'):
##         OptMethod.__init__(self, name, my_fmin_powell)

## class StoGo(OptMethod):
##     def __init__(self, name='stogo'):
##     OptMethod.__init__(self, name, stogo)

###############################################################################

As you can see: one of us has already tried all the optimization methods
shipped with scipy but that was a long time ago.  I encourage people to
hook up the scipy routines and test them (we can certainly help you if you
have any questions :)

dan

PS There is a package called NLOPT,
http://ab-initio.mit.edu/wiki/index.php/NLopt, that is maintained by Steven
Johnson (I think he is one of the authors of FFTW) that is worth trying.  I
have not tried to hook up the NLOPT library but I have tried a few of
methods (newuoa and StoGo, chokkan lbfgs) that is shipped with it though.

I believe the optimization which I called PortChi and PortFct were the
methods that was initially shipped on the GLAST mission before they gave up
and used minuit etc....

**Comment by dtnguyen2** [comment]
done

**Comment by DougBurke** [comment]
@dtnguyen2 Does "done" refer to your "feature/scipymin" beanch - e.g. commit 45c4b625ecf9ac84a7db18131a710b6fa6048ab9 ?

**Comment by dtnguyen2** [comment]
yes

**Comment by matteobachetti** [comment]
Thanks! This is great!

**Comment by DougBurke** [comment]
@hamogu - I think with #2303 merged we can close this issue, but just wanted to check.

**Comment by hamogu** [comment]
Correct. That's implemented, I just forgot to link this issue to auto-close. Thanks @matteobachetti for suggesting it!
